homeassistant/components/fritz/binary_sensor.py

```diff
@@ -17,79 +24,91 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def _retrieve_connection_state(status: FritzStatus) -> bool:
+    """Return download line attenuation."""
+    return bool(status.is_connected)
+
+
+def _retrieve_link_state(status: FritzStatus) -> bool:
+    """Return download line attenuation."""
+    return bool(status.is_linked)
+
+
+@dataclass
+class SensorData(SensorEntityDescription):
+    """Fritz sensor data class."""
+
+    state_provider: Callable | None = None
+
+
+SENSOR_DATA = {
+    "is_connected": SensorData(
+        key="is_connected",
+        name="Connection",
+        device_class=DEVICE_CLASS_CONNECTIVITY,
+        state_provider=_retrieve_connection_state,
+    ),
+    "is_linked": SensorData(
+        key="is_linked",
+        name="Link",
+        device_class=DEVICE_CLASS_PLUG,
+        state_provider=_retrieve_link_state,
+    ),
+}
```

Comment:
The dict key and the description key is identical, use a tuple here instead of a dict

Reply:
Done